### PR TITLE
Issue/2191 display product reviews

### LINKED ISSUE #2191: Edit Products M3 - Product Reviews: Option to enable/disable product reviews
As part of the M3 release, we should provide an option to enable/disable product reviews from the Product Settings page. If product reviews are switched on, we should display the product review count in the Product Detail screen.

.

### PR DESCRIPTION
**Pull request by anitaa1990**
Fixes #2191 by adding logic to display reviews for a product.

#### Changes
- Added logic to display the reviews count for a product, if reviews are enabled.
- Created a new fragment, module and View Model to handle the changes needed to show only the reviews for a specific product.

#### Screenshots
. . . 

#### Testing
- Click on a product with no reviews enabled.
  - Click on the Product Settings option from the Product Detail menu.
  - Enable reviews for the product and click on `DONE`.
  - Verify that the review count for the product is displayed.
  - Click on the Reviews section.
  - Verify that the approved reviews for the product are displayed correctly.
  - From the web, add a new review on the selected product, and approve it.
  - Pull to refresh the Reviews screen. The new review should appear on top.
- Click on a product with reviews enabled, but no reviews available.
  - Click on the reviews section and verify that an empty view is displayed correctly.
  - From the web, add a new review on the selected product, and approve it.
  - Pull to refresh the Reviews screen. The new review should appear on the screen and the empty view should disappear.
- Click on a product with reviews enabled, and atleast 40 reviews (to test pagination).
  - Click on the reviews section and verify that pagination works as expected.

~**This PR is in draft till this [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1659) and this [product type PR](https://github.com/woocommerce/woocommerce-android/pull/2712) can be reviewed and merged.**~

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt [hunk 1]
```diff
@@ -300,6 +301,11 @@ class AnalyticsTracker private constructor(private val context: Context) {
         PRODUCT_TAGS_LOAD_FAILED,
         PRODUCT_TAGS_PULLED_TO_REFRESH,
 
+        // -- Product reviews
+        PRODUCT_REVIEWS_LOADED,
+        PRODUCT_REVIEWS_LOAD_FAILED,
+        PRODUCT_REVIEWS_PULLED_TO_REFRESH,
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt:307]: NP: I didn't find these in the spreadsheet. You probably forgot to add them.

**Review by 0nko — COMMENTED** [review]
Nice job! Everything works as expected. I just noticed that when I tap on a review item in the list nothing happens. Is that expected?

**Comment by anitaa1990** [comment]
Thanks for the review @0nko!!

>  I just noticed that when I tap on a review item in the list nothing happens. Is that expected?

Yup, that's correct 👍  I think we discussed and decided not to display a detail screen for reviews from the Product tab

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt (hunk 1 above)
Comment (anitaa1990) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt:307]: Good point! I added those events to the spreadsheet. Thanks for the reminder 😄

**Review by 0nko — APPROVED** [review]
:shipit:

**Pull request merged**